### PR TITLE
feat: added the gitlab cleanup code

### PR DESCRIPTION
**Pull request by emamihe**
this golang code will cleanup the github stalled runners

### CODE REVIEW

File: scripts/gitlab_cleanup/main.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+    "k8s.io/client-go/kubernetes"
+    "k8s.io/client-go/rest"
+)
+
+func main() {
+    config, err := rest.InClusterConfig()
+    if err != nil {
+        fmt.Printf("Error creating in-cluster config: %s\n", err)
+        os.Exit(1)
```
Comment (l0r1s) [inline scripts/gitlab_cleanup/main.go:18]:
```suggestion
        log.Fatalf("Error creating in-cluster config: %s\n", err)
```

File: scripts/gitlab_cleanup/main.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    clientset, err := kubernetes.NewForConfig(config)
+    if err != nil {
+        fmt.Printf("Error creating clientset: %s\n", err)
+        os.Exit(1)
```
Comment (l0r1s) [inline scripts/gitlab_cleanup/main.go:24]:
```suggestion
        log.Fatalf("Error creating clientset: %s\n", err)
```

File: scripts/gitlab_cleanup/main.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    pods, err := clientset.CoreV1().Pods("gitlab").List(context.Background(), metav1.ListOptions{
+        LabelSelector: "app=gitlab,selector=runner",
+    })
+    if err != nil {
+        fmt.Printf("Error listing pods: %s\n", err)
+        os.Exit(1)
```
Comment (l0r1s) [inline scripts/gitlab_cleanup/main.go:32]:
```suggestion
        log.Fatalf("Error listing pods: %s\n", err)
```

**Review by l0r1s — APPROVED** [review]
Just a few nits but looks good 👍

File: scripts/gitlab_cleanup/main.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    for _, pod := range pods.Items {
+        if time.Since(pod.CreationTimestamp.Time).Hours() > 12 {
```
Comment (l0r1s) [inline scripts/gitlab_cleanup/main.go:36]: We could probably extract a const, maybe having it in the env to make this configurable without having to change the code ? Wdyt ?

Comment (pepoviola) [reply scripts/gitlab_cleanup/main.go:36]: Yes, I think an env var with a default value could be better.

File: scripts/gitlab_cleanup/main.go [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    if err != nil {
+        log.Fatalf("Error creating clientset: %s\n", err)
+    }
+
+    podAgeThresholdHours := 12 // default
+    if envVal, exists := os.LookupEnv("POD_AGE_THRESHOLD_HOURS"); exists {
+        if val, err := strconv.Atoi(envVal); err == nil {
+            podAgeThresholdHours = val
+        }
+    }
```
Comment (l0r1s) [inline scripts/gitlab_cleanup/main.go:33]: 👍

**Review by l0r1s — APPROVED** [review]

**Pull request merged**